lib/discourse_api/api/groups.rb

```diff
@@ -107,11 +107,14 @@ def delete_group(group_id)
       end
 
       def group_members(group_name, params = {})
```

Comment:
I think it would be better to add a third parameter here instead of assigning the entire params hash to options.

so something like `group_members(group_name, params = { }, all = false)`

Reply:
I thought about that, but then that would make params a required argument.

You would have to do something like this `group_members("foo", {}, true)`. I think its better to just make it optional

Reply:
I really think it is fine adding a third param. It is still optional and not a breaking change if you just stick with the default `group_members("foo")`.

I've done a lot of cleanup to make sure all params are listed either in the `.optional`, `.default`, or `.required` method, so I don't really like just throwing a hidden one in there. The problem with the current `API.params` design is that it would send the entire hash in the payload and we don't want to send extra params to the discourse api otherwise I would say just add it to the `.optional` method.

Reply:
I think its probably a good idea to make an exception in this case.

> I've done a lot of cleanup to make sure all params are listed either in the .optional, .default, or .required method
This still does its job. the API class responsibly controls what optional and default arguments can only be passed to the Discourse API.

>so I don't really like just throwing a hidden one in there
Either way the "all" argument is going to be a hidden argument/nuance of the method. Adding it as a 3rd argument means now there's a hidden "gotcha" where you have to pass in an empty hash as a second argument. `group_members("foo", {}, true)`

your current suggestion means someone would have to call `group_members("foo", {}, true)`  to get all the users. This requires an empty hash in the 2nd argument, thats an even worse